cores/esp32/esp32-hal-i2c.c

```diff
@@ -1611,7 +1612,7 @@ i2c_err_t i2cSetFrequency(i2c_t * i2c, uint32_t clk_speed)
     }
     I2C_FIFO_CONF_t f;
   
-    uint32_t period = (APB_CLK_FREQ/clk_speed) / 2;
+    uint32_t period = (rtc_clk_apb_freq_get()/clk_speed) / 2;
```

Comment:
I tried this mod a couple of weeks ago, the returned value didn't change, it stayed at 80,000,000.  But the actual hardware clock changed speed, I could see the i2c bus speed change on my oscilloscope.

Have you changed the code inside `rtc_clk_apb_freq_get()` since Rc1 1.0.1 ?

Reply:
to me it changes now when you set the cpu frequency below 80MHz. I have not tested I2C, but tested other peripherals. Give it a go :)

Reply:
see private glitter

Reply:
An upper bounds needs to be applied to I2C bus freq versus CPU freq:
```c++
#define MIN_I2C_CLKS 100
    uint32_t period = (rtc_clk_apb_freq_get()/clk_speed) / 2;
    if(period < (MIN_I2C_CLKS/2) ){
      period = (MIN_I2C_CLKS/2);
      log_w("APB Freq too slow, Reducing i2c Freq to %d Hz",rtc_clk_apb_freq_get()/(period*2));
    }      
```
This limits it to 1/100th APB frequency so a 5MHz APB has a max i2c of 50KHz down to 10KHz at 1MHz.  I tried up to 1/80 APB, I2C did not function.  In my newer code I have other places I need `MIN_I2C_CLKS`.

Reply:
This is a good solution I think. Will also venture into dynamic cpu scaling and will see how we can manage that.